### PR TITLE
Merge template

### PR DESCRIPTION
**Pull request by myluki2000**
## Description of changes
Merges changes from the template and implements API changes required by the GraphQL Mesh Gateway. No direct issue exists for this, but it is required by https://github.com/IT-REX-Platform/graphql_gateway/issues/4
  
## How has this been tested?
Please describe the test strategy you followed.

- [ ] automated unit test
- [ ] automated integration test
- [ ] automated acceptance test
- [x] manual, exploratory test

1. Create multiple new media records using the appropriate GraphQL query.
2. Query for media records using GraphQL
3. Query for media records by ID using GraphQL
4. Query for media records by content ID using GraphQL
5. Update a record
6. Delete a record

These are rudamentary tests, proper automated testing will be added shortly as soon as the API schema of the service has settled down and everything has been tested and ensured appropriately functioning together with the GraphQL Mesh gateway.
    
## Checklist before requesting a review
- [x] My code follows the [coding guidelines](https://github.com/IT-REX-Platform/wiki/blob/main/dev-manuals/backend/coding-guidelines.md) of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] My code fulfilles all acceptance criteria
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] I have made corresponding changes to the documentation
- [ ] I have added explanation of architectural decision and rationales to [wiki/adr](https://github.com/IT-REX-Platform/wiki/tree/main/adr)
- [ ] I have updated the changes in the ticket description

## Checklist for reviewer
- [x] The code works and does not throw errors
- [x] The code is easy to understand and there are no confusing parts
- [x] The code follows the [coding guidelines](https://github.com/IT-REX-Platform/wiki/blob/main/dev-manuals/backend/coding-guidelines.md) of this project
- [x] The code change accomplishes what it is supposed to do
- [x] I cannot think of any use case in which the code does not behave as intended
- [ ] The added and existing tests reasonably cover the code change
- [ ] I cannot think of any test cases, input or edge cases that should be tested in addition
- [x] Description of the change is included in the documentation

### CODE REVIEW

File: src/main/resources/graphql/schema.graphqls [hunk 1]
```diff
@@ -63,16 +62,25 @@ enum MediaType {
...
   mediaRecordsById(ids: [UUID!]!): [MediaRecord!]!
+  # Returns all media records
   mediaRecords: [MediaRecord!]!
+  # Returns the media records associated the given content IDs as a list of lists where each sublist contains
+  # the media records associated with the content ID at the same index in the input list
+  mediaRecordsByContentIds(contentIds: [UUID!]!): [[MediaRecord!]!]!
 }
 
 type Mutation {
+  # Creates a new media record
```
Comment (ZombieAlienRobot) [inline src/main/resources/graphql/schema.graphqls:75]: Aren't these comments useless? The Mutations are pretty selfexplanatory. Imo they just create visual clutter in the File.
On the other hand it is pretty nice that postman shows the comments.

File: src/main/java/de/unistuttgart/iste/gits/media_service/controller/MediaController.java [hunk 2]
```diff
@@ -34,6 +34,10 @@ public List<MediaRecordDto> mediaRecordsById(@Argument List<UUID> ids) {
         return mediaService.getMediaRecordsById(ids);
     }
 
+    @QueryMapping List<List<MediaRecordDto>> mediaRecordsByContentIds(@Argument List<UUID> contentIds) {
```
Comment (PaulBredl) [inline src/main/java/de/unistuttgart/iste/gits/media_service/controller/MediaController.java:37]: (No todo)
Usually annotations are on a seperate line (but I don't care about such things)

**Review by PaulBredl — APPROVED** [review]
I don't have the time to run the manual test but I believe you that it works. 
I have a remark on the repository, but if it works now like this it is fine. Just technically I don't think you actually need the explicit SQL.

**Review by ZombieAlienRobot — APPROVED** [review]

File: src/test/java/de/unistuttgart/iste/gits/media_service/integration/QueryMediaTest.java [hunk 3]
```diff
@@ -0,0 +1,8 @@
+package de.unistuttgart.iste.gits.media_service.integration;
+
+import de.unistuttgart.iste.gits.util.GraphQlApiTest;
+
+@GraphQlApiTest
+public class QueryMediaTest {
```
[the remark this answers is not in the document]
Comment (myluki2000) [reply src/test/java/de/unistuttgart/iste/gits/media_service/integration/QueryMediaTest.java:6]: It's there to remind me to write tests soon :D

File: src/main/resources/graphql/schema.graphqls (hunk 1 above)
Comment (myluki2000) [reply src/main/resources/graphql/schema.graphqls:75]: We auto-generate documentation from the graphql schema. The doc comments are a bit bare-bones for now but for some things I think there definitely will be some notes etc. that will need to be added to the doc comments.

File: src/main/java/de/unistuttgart/iste/gits/media_service/controller/MediaController.java (hunk 2 above)
Comment (myluki2000) [reply src/main/java/de/unistuttgart/iste/gits/media_service/controller/MediaController.java:37]: Blame copilot lol. Fixed it.

File: src/main/java/de/unistuttgart/iste/gits/media_service/persistence/repository/MediaRecordRepository.java [hunk 4]
```diff
@@ -1,15 +1,17 @@
...
+import java.util.List;
 import java.util.Optional;
 import java.util.UUID;
 
 @Repository
 public interface MediaRecordRepository extends JpaRepository<MediaRecordEntity, UUID> {
 
-    Optional<MediaRecordEntity> findByName(String name);
-
+    @Query("SELECT DISTINCT media FROM MediaRecord media JOIN media.contentIds actualIds WHERE actualIds IN :contentIds")
```
[the remark this answers is not in the document]
Comment (PaulBredl) [reply src/main/java/de/unistuttgart/iste/gits/media_service/persistence/repository/MediaRecordRepository.java:15]: Could me explored in the next sprint i guess

File: docker-compose.yml [hunk 5]
```diff
@@ -49,9 +49,9 @@ services:
     restart: always
     network_mode: bridge
     expose:
-      - 1033
+      - 3033
     ports:
-      - "1033:5432"
+      - "3033:5432"
```
Comment (PaulBredl) [inline docker-compose.yml:54]: If you change that here, you also need to change it in GraphQlApiTest annotation

**Review by PaulBredl — CHANGES_REQUESTED** [review]

File: docker-compose.yml (hunk 5 above)
Comment (PaulBredl) [reply docker-compose.yml:54]: ok too late xD but that can also be done later

Comment (myluki2000) [reply docker-compose.yml:54]: That's what I did. Check in the annotation file ;)
